Modules/_sqlite/connection.c

```diff
@@ -1442,6 +1457,169 @@ pysqlite_connection_iterdump(pysqlite_Connection* self, PyObject* args)
     return retval;
 }
 
+#ifdef HAVE_BACKUP_API
+static PyObject *
+pysqlite_connection_backup(pysqlite_Connection *self, PyObject *args, PyObject *kwds)
+{
+    PyObject *target = NULL;
+    const char *filename;
+    int pages = -1;
+    PyObject *progress = Py_None;
+    char *name = "main";
+    int rc;
+    int cberr = 0;
+    double sleep_secs = 0.250;
+    sqlite3 *bckconn;
+    sqlite3_backup *bckhandle;
+    static char *keywords[] = {"target", "pages", "progress", "name", "sleep", NULL};
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "O|$iOsd:backup", keywords,
+                                     &target, &pages, &progress, &name, &sleep_secs)) {
+        return NULL;
+    }
+
+    if (!PyUnicode_Check(target) && !PyObject_TypeCheck(target, &pysqlite_ConnectionType)) {
+        PyErr_SetString(PyExc_TypeError,
+                        "target argument must be either a string or a Connection instance");
+        return NULL;
+    }
+
+    if (progress != Py_None && !PyCallable_Check(progress)) {
+        PyErr_SetString(PyExc_TypeError, "progress argument must be a callable");
+        return NULL;
+    }
+
+    if (pages == 0) {
+        pages = -1;
+    }
+
+    if (PyUnicode_Check(target)) {
+        filename = PyUnicode_AsUTF8(target);
+        if (filename == NULL) {
+            return NULL;
+        }
+
+        Py_BEGIN_ALLOW_THREADS
+        rc = sqlite3_open(filename, &bckconn);
+        Py_END_ALLOW_THREADS
+
+        if (rc != SQLITE_OK) {
```

Comment:
`pysqlite_connection_init()` uses a different code for handling errors. Is there an error in `pysqlite_connection_init()` or the code here can be simpler?

Reply:
I think a separate issue should be open for fixing `pysqlite_connection_init()`. Since this is a bug fix, it needs to be backported to other branches.

Reply:
See [#32274](https://bugs.python.org/issue32274): I was wrong about that, the database connection will be released by the `pysqlite_connection_dealloc()` function in any case.